### ISSUE TITLE
SmartOS/OmniOS - mount module fails.

### ISSUE THREAD

**Issue by johngrasty** [saltstack/salt#14915]
https://github.com/saltstack/salt/blob/develop/salt/modules/mount.py#L36

The mount module runs `mount -l` on line 36 (of the develop branch, 2014.1 does the same though) in the _list_mounts function. `-l` is an illegal option on SmartOS/OmniOS (and I assume other illumos distros. 

I believe that it is used for returning the disk labels in [linux](http://manpages.debian.org/cgi-bin/man.cgi?query=mount&apropos=0&sektion=0&manpath=Debian+7.0+wheezy&format=html&locale=en). In FreeBSD, `-l` is used "used in conjunction with the -a option, also mount those file systems which are marked as “late”." While it does no harm, it might be best to not use it on FreeBSD as well.

Here is the error output on the minion's logs:

``````
[ERROR   ] Command 'mount -l' failed with return code: 1
[ERROR   ] stderr: mount: illegal option -- l
Usage:
mount [-v | -p]
mount [-F FSType] [-V] [current_options] [-o specific_options]
        {special | mount_point}
mount [-F FSType] [-V] [current_options] [-o specific_options]
        special mount_point
mount -a [-F FSType ] [-V] [current_options] [-o specific_options]
        [mount_point ...]
[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "salt/state.py", line 1379, in call
  File "salt/states/mount.py", line 87, in mounted
  File "salt/modules/mount.py", line 108, in active
  File "salt/modules/mount.py", line 30, in _active_mountinfo
  File "salt/modules/mount.py", line 25, in _list_mounts
IndexError: list index out of rang
`````````

**Comment by ssgward** [comment]
@johngrasty - is this still an issue? This was logged long ago. Wondering if the latest Salt still shows the problem.

**Comment by johngrasty** [comment]
Yes. It appears that the code has not been changed. I think there would also be significant problems with other parts the mount module for illumos based distributions. I tested many of the ones available with 100% failure rate.

**Comment by johngrasty** [comment]
This may be a start, though I am not a programmer.

```
def _list_mounts():
    ret = {}
    if __grains__['os'] in ['MacOS', 'Darwin']:
        mounts = __salt__['cmd.run_stdout']('mount')
    if __grains__['os_family'] in ['Solaris']:
        mounts = __salt__['cmd.run_stdout']('mount')
    else:
        mounts = __salt__['cmd.run_stdout']('mount -l')

    for line in mounts.split('\n'):
        comps = re.sub(r"\s+", " ", line).split()
        ret[comps[2]] = comps[0]
    return ret
```

**Comment by sjorge** [comment]
@sjorge tagging my self, I can replicate.
Will have a look once I have a bit more free time.

**Comment by rallytime** [comment]
@johngrasty How does the fix in #34254 look to you?

**Comment by rallytime** [comment]
This should be fixed by #34254. Since we haven't heard back, I am closing this. If this pops up again, leave a comment and we can re-open and take another look. Thanks all!
